demos/vlm_npu/README.md

```diff
@@ -0,0 +1,252 @@
+  "model": "microsoft/Phi-3.5-vision-instruct",
+  "object": "chat.completion",
+  "usage": {
+    "prompt_tokens": 19,
+    "completion_tokens": 83,
+    "total_tokens": 102
+  }
+}
+```
+:::
+
+:::{dropdown} **Unary call with OpenAI Python package**
+
+The endpoints `chat/completions` are compatible with OpenAI client so it can be easily used to generate code also in streaming mode:
+
+Install the client library:
+The endpoints `chat/completions` are compatible with OpenAI client so it can be easily used to generate code also in streaming mode:
+
+Install the client library:
+```console
+pip3 install openai
+```
+```python
+from openai import OpenAI
+import base64
+base_url='http://localhost:8080/v3'
+model_name = "OpenGVLab/InternVL2_5-8B"
+
+client = OpenAI(api_key='unused', base_url=base_url)
+
+def convert_image(Image):
+    with open(Image,'rb' ) as file:
+        base64_image = base64.b64encode(file.read()).decode("utf-8")
+    return base64_image
+
+stream = client.chat.completions.create(
+    model=model_name,
+    messages=[
+        {
+            "role": "user",
+            "content": [
+              {"type": "text", "text": "Describe what is one the picture."},
+              {"type": "image_url", "image_url": {"url": f"data:image/jpeg;base64,{convert_image('zebra.jpeg')}"}}
+            ]
+        }
+        ],
+    stream=True,
+)
+for chunk in stream:
+    if chunk.choices[0].delta.content is not None:
+        print(chunk.choices[0].delta.content, end="", flush=True)
+```
+
+Output:
+```
+The picture features a zebra standing in a grassy area. The zebra is characterized by its distinctive black and white striped pattern, which covers its entire body, including its legs, neck, and head. Zebras have small, rounded ears and a long, flowing tail. The background appears to be a natural grassy habitat, typical of a savanna or plain.
+```
+
+:::
+
+## Benchmarking text generation with high concurrency
+
+OpenVINO Model Server with NPU acceleration process the requests sequentially. For that reason, benchmarking should be performed in max_concurrency set to 1.
+Parallel requests will be accepted but they will wait for a free execution slot.
+Benchmarking can be demonstrated using benchmarking app from vLLM repository:
+```console
+git clone --branch v0.7.3 --depth 1 https://github.com/vllm-project/vllm
+cd vllm
+pip3 install -r requirements-cpu.txt --extra-index-url https://download.pytorch.org/whl/cpu
+cd benchmarks
+curl -L https://huggingface.co/datasets/anon8231489123/ShareGPT_Vicuna_unfiltered/resolve/main/ShareGPT_V3_unfiltered_cleaned_split.json -o ShareGPT_V3_unfiltered_cleaned_split.json # sample dataset
+python benchmark_serving.py --backend openai-chat --dataset-name hf --dataset-path lmarena-ai/vision-arena-bench-v0.1 --hf-split train --host localhost --port 8000 --model OpenGVLab/InternVL2_5-8B --endpoint /v1/chat/completions  --request-rate 1 --num-prompts 10 --trust-remote-code --max-concurrency 1
```

Comment:
fails with error:
Traceback (most recent call last):
  File "/opt/home/k8sworker/ngroza/vllm/benchmarks/benchmark_serving.py", line 1330, in <module>
    main(args)
  File "/opt/home/k8sworker/ngroza/vllm/benchmarks/benchmark_serving.py", line 959, in main
    benchmark_result = asyncio.run(
  File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/opt/home/k8sworker/ngroza/vllm/benchmarks/benchmark_serving.py", line 601, in benchmark
    raise ValueError(
ValueError: Initial test run failed - Please make sure benchmark arguments are correctly specified. Error: Bad Request